### PR TITLE
Add call-graph overview

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

**Review by chrisr-diffblue — APPROVED** [review]
LGTM

File: src/analyses/call_graph.h [hunk 1]
```diff
@@ -18,6 +18,23 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <goto-programs/goto_model.h>
 #include <util/graph.h>
 
+/// A call graph (https://en.wikipedia.org/wiki/Call_graph) for a GOTO model
+/// or GOTO functions collection.
+///
+/// The public constructors build a complete call graph, while
+/// \ref create_from_root_function can be used to create a partial call graph
```
Comment (peterschrammel) [inline src/analyses/call_graph.h]: Is this ref resolved correctly?

**Comment by polgreen** [comment]
Michael asked me to comment on this; LGTM

File: src/analyses/call_graph.h (hunk 1 above)
Comment (smowton) [reply src/analyses/call_graph.h]: fixed

**Pull request merged**